game-app/game-core/src/main/java/games/strategy/triplea/attachments/RulesAttachment.java

```diff
@@ -1064,6 +1104,18 @@ private boolean checkTechs(final GamePlayer player, final TechnologyFrontier tec
     return found >= techCount;
   }
 
+  @VisibleForTesting
+  public boolean checkHasResource(final List<GamePlayer> players) {
+    int itotal = 0;
```

Comment:
Same: Follow [G1. Choose variable names that provide useful information](https://openjdk.org/projects/amber/guides/lvti-style-guide#G1)